### DISCUSSION TITLE
AxAXISMEDIACONTROLLLib

### DISCUSSION THREAD

**Discussion by STWIGGDWP** [AxisCommunications/discussions#40] [Axis Windows SDKs] [answered]
I am attempting to build the AXISMediaControlSDK_7_42 C# sample on a Windows 11 PC with Visual Studio 2022.

However, even after installing the SDK, and setting the visual studio build for Win32 -- when attempting to open the VideoForm.cs in any way -- I keep getting the following error :  Class not registered (Exception from HRESULT: 0x80040154 (REGDB_E_CLASSNOTREG))
![image](https://user-images.githubusercontent.com/111086503/184226215-3f2b100c-1dc4-480d-8122-bf31a089bb4d.png)

Any help would be appreciated, thanks!

**Comment by fre3de** [comment, marked as answer]
Hi,

.net 5 and later doesn't have support for ActiveX controls in the designer. AMC will work if you add it dynamically to the code, but you cannot open the designer. For the designer to work, you have to use .net 4.8. 

More info: https://developercommunity.visualstudio.com/t/support-for-activex-controls-in-windows-forms-desi/1435698

/Fredrik

**Reply by STWIGGDWP** [reply]
I am using 4.8
![image](https://user-images.githubusercontent.com/111086503/184706664-f301bce7-b45c-4ecf-b854-7496a5cbd54d.png)

**Reply by fre3de** [reply]
Have you installed the 64 bit SDK and/or 32 bit? The designer in Visual Studio can now run 64 bit which means it's no longer necessary to have the 32 bit AMC SDK installed. 

Does it work if you create a new project in VS 2022? This video shows how that can be done: https://youtu.be/vef_UFheSTE

**Reply by STWIGGDWP** [reply]
I have installed the 32-bit SDK.
I will give the 64 bit installer a try, and let you know, thanks.

**Reply by STWIGGDWP** [reply]
Ok that worked, thanks.
Please update your website to reflect this change:
![image](https://user-images.githubusercontent.com/111086503/184913335-2b3d0fe7-2b5f-4f65-a9af-7685721a25bc.png)

**Reply by fre3de** [reply]
Good to hear you got it working! Thanks for pointing out that the note should be updated.

**Comment by ntdgo** [comment]
Hi @fre3de,

Can we add dll reference without install SDK?
My program cannot run on the computer which does not install SDK.

**Reply by fre3de** [reply]
Hi, 

Yes, Axis Media Control SDK supports registration-free installation.
Read more about registration free installation in the SDK documentation under Introducing AXIS Media Control - Redistributing AMC - Registration free installation.
You find the AMC SDK documentation in SDK installation folder\doc

Best regards
Fredrik
